middleware/limiter/limiter.go

```diff
@@ -68,18 +67,24 @@ func New(config ...Config) fiber.Handler {
 
 		} else if ts >= e.exp {
 			// Check if entry is expired
-			e.hits = 0
+			e.prevHits = e.currHits
```

Comment:
`e.prevHits = e.currHits` only happens when `ts == e.exp`, otherwise `e` was GCed.

Reply:
Fixed this by setting the garbage collection expiry to last 2 windows. There was also an issue with the expiry not being set correctly. It was being set to `time of first request in new window + expiration` when it should be 
`time window started + expiration`.